consensus/src/miner.rs

```diff
@@ -97,106 +92,116 @@ impl<S: Storage> Miner<S> {
             }
         }
 
-        let (records, tx) = self.consensus.create_coinbase_transaction(
-            self.consensus.ledger.get_current_block_height() + 1,
-            transactions,
-            program_vk_hash,
-            new_birth_programs,
-            new_death_programs,
-            self.address.clone(),
-            rng,
-        )?;
-
-        transactions.push(tx);
-        Ok(records)
+        //(records, tx)
+        let response = self
+            .consensus
+            .create_coinbase_transaction(
+                self.block_height as u32 + 1,
+                transactions,
+                program_vk_hash,
+                new_birth_programs,
+                new_death_programs,
+                vec![self.address.clone(); Components::NUM_OUTPUT_RECORDS]
+                    .into_iter()
+                    .map(|x| x.into())
+                    .collect(),
+            )
+            .await?;
+
+        transactions.push(response.transaction);
+        Ok(response.records)
     }
 
     /// Acquires the storage lock and returns the previous block header and verified transactions.
     #[allow(clippy::type_complexity)]
-    pub fn establish_block(
+    pub async fn establish_block(
         &self,
-        transactions: &DPCTransactions<Tx>,
-    ) -> Result<(BlockHeader, DPCTransactions<Tx>, Vec<DPCRecord<Components>>), ConsensusError> {
-        let rng = &mut thread_rng();
-        let mut transactions = transactions.clone();
-        let coinbase_records = self.add_coinbase_transaction(&mut transactions, rng)?;
+        transactions: &[SerialTransaction],
```

Comment:
I looked at the callsite and it should be possible to pass these as a `Vec`, removing the need for the `to_vec` below.